validations/src/validations.rs

```diff
@@ -1677,6 +1754,82 @@ pub fn validate_block_transactions(
         );
     }
 
+    // validate stake transactions in a block
+    let mut st_mt = ProgressiveMerkleTree::sha256();
+    let mut st_weight: u32 = 0;
+
+    // Check if the block contains more than one stake tx from the same operator
+    for i in 1..block.txns.stake_txns.len() {
+        let found = block.txns.stake_txns[i..].iter().find(|stake_tx| {
+            let stake_tx_aux = &block.txns.stake_txns[i - 1];
+
+            stake_tx.body.output.authorization.public_key
+                == stake_tx_aux.body.output.authorization.public_key
+        });
+
+        // TODO: refactor
+        if found.is_some() {
+            return Err(BlockError::RepeatedStakeOperator {
+                pkh: found
+                    .unwrap()
+                    .body
+                    .output
+                    .authorization
+                    .public_key
+                    .pkh()
+                    .hash(),
+            }
+            .into());
+        }
+    }
+
+    for transaction in &block.txns.stake_txns {
+        let (inputs, _output, fee, weight, change) = validate_stake_transaction(
+            transaction,
+            &utxo_diff,
+            epoch,
+            epoch_constants,
+            signatures_to_verify,
+        )?;
```

Comment:
Inconsistent use of `?` and `return` syntax in the same function. Let's decide for one or another. See my reasoning in my other comments.